core/primitives/transaction_validity.hpp

```diff
@@ -102,14 +102,16 @@ namespace kagome::primitives {
   template <class Stream,
             typename = std::enable_if_t<Stream::is_encoder_stream>>
   Stream &operator<<(Stream &s, const InvalidTransaction &v) {
-    return s << static_cast<uint8_t>(v);
+    return s << static_cast<uint8_t>(v)
+                    - 1;  // -1 is needed for compatibility with Rust
```

Comment:
Multiline suggestion:
    // -1 is needed for compatibility with Rust; indices of error codes start from 0 there, while in kagome they must start from 1 because of std::error_code policy
    return s << static_cast<uint8_t>(v) - 1; 

And the corresponding comment in the decoding of this enum and encoding-decoding of unknown transaction enum. I believe such comment may be helpful in future, as this whole issue with codes turned out to be pretty non-trivial.